### PR TITLE
chore: add engine clause to package.json

### PR DESCRIPTION
**Pull request by srl295**
- will warn if used in node !18
- per https://docs.npmjs.com/cli/v6/configuring-npm/package-json#engines this could be an error if `engine-strict` was enabled. 

```console
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: undefined,
npm WARN EBADENGINE   required: { node: '^18.x' },
npm WARN EBADENGINE   current: { node: 'v16.20.0', npm: '8.19.4' }
npm WARN EBADENGINE }
```

@keymanapp-test-bot skip

#8950

### CODE REVIEW

**Comment by srl295** [comment]
it will warn if you're on 19 also, so we'll be prompted to fix when ever we bump again.

**Comment by srl295** [comment]
I made it 18.x to be generic and not try to give a specific version.

**Review by mcdurdin — APPROVED** [review]
LGTM -- we should probably check and if necessary refresh the docs/build files at the same time?

**Comment by srl295** [comment]
> LGTM -- we should probably check and if necessary refresh the docs/build files at the same time?

you mean, whenever this changes again?

I don't think this change necessitates anything, just causes a fail-fast

**Comment by mcdurdin** [comment]
I mean where we are non-specific about Node version in the current build config docs -- e.g. docs/build/macos.md (and seems linux.md doesn't even mention it but we need it for hextobin for core). Just seems a good trailer to tag onto this patch.

**Comment by srl295** [comment]
@mcdurdin will take a look.

**Comment by srl295** [comment]
@ermshiperete @mcdurdin will we need node.js in the debian/control file?

**Comment by ermshiperete** [comment]
> will we need node.js in the debian/control file?

This would mean that we'd have to repackage node.js 18 for Ubuntu 20.04 Focal and 22.04 Jammy. 

22.10 Kinetic and later come with node.js 18 in the default repos.

**Comment by mcdurdin** [comment]
> will we need node.js in the debian/control file?

No, because Keyman for Linux doesn't use Node.

File: docs/build/linux-ubuntu.md [hunk 1]
```diff
@@ -49,6 +49,10 @@ sudo apt install devscripts equivs
 sudo mk-build-deps --install linux/debian/control
 ```
 
+### Node.js
+
+Node.js v18 is required for the core build.
```
Comment (mcdurdin) [inline docs/build/linux-ubuntu.md:54]:
```suggestion
Node.js v18 is required for Core build, Web tests, and Developer command line tools.
```

I think I'd like to spend a day on the dev docs at some point to get them consistent once more -- the Win/mac docs also need this info...

**Review by mcdurdin — APPROVED** [review]

**Pull request merged**